src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java

```diff
@@ -55,25 +68,24 @@ public String requestInstance(ComputeOrder computeOrder, AzureUser azureUser) th
         LOGGER.info(Messages.Info.REQUESTING_INSTANCE_FROM_PROVIDER);
 
         String networkInterfaceId = getNetworkInterfaceId(computeOrder, azureUser);
-        String virtualMachineSizeName = getVirtualMachineSizeName(computeOrder, azureUser);
+        VirtualMachineSize virtualMachineSize = getVirtualMachineSize(computeOrder, azureUser);
         int diskSize = AzureGeneralPolicy.getDisk(computeOrder);
-        AzureGetImageRef azureVirtualMachineImage = AzureImageOperationUtil
-                .buildAzureVirtualMachineImageBy(computeOrder.getImageId());
-        String virtualMachineName = AzureInstancePolicy
-                .generateAzureResourceNameBy(computeOrder, azureUser);
-        String userData = getUserData(computeOrder);
-        String osUserName = computeOrder.getId();
-        String osUserPassword = AzureGeneralPolicy.generatePassword();
-        String osComputeName = computeOrder.getId();
+        String imageId = computeOrder.getImageId();
+        AzureGetImageRef imageRef = AzureImageOperationUtil.buildAzureVirtualMachineImageBy(imageId);
         String regionName = this.defaultRegionName;
         String resourceGroupName = this.defaultResourceGroupName;
+        String virtualMachineName = AzureInstancePolicy.defineAzureResourceName(computeOrder, azureUser, resourceGroupName);
+        String userData = getUserData(computeOrder);
+        String osUserName = DEFAULT_OS_USER_NAME;
+        String osUserPassword = AzureGeneralPolicy.generatePassword();
+        String osComputeName = computeOrder.getName();
 
         AzureCreateVirtualMachineRef azureCreateVirtualMachineRef = AzureCreateVirtualMachineRef.builder()
                 .virtualMachineName(virtualMachineName)
-                .azureGetImageRef(azureVirtualMachineImage)
+                .azureGetImageRef(imageRef)
                 .networkInterfaceId(networkInterfaceId)
                 .diskSize(diskSize)
-                .size(virtualMachineSizeName)
+                .size(virtualMachineSize.name())
```

Comment:
Extract it to a variable in order to follow the pattern.

```suggestion
String size = virtualMachineSize.name();
...
                .size(size)
...
```

Reply:
Ok, I get it.